internal/forms/forms.go

```diff
@@ -705,6 +710,99 @@ func (m *Manager) findAbsPathForTemplatePath(tmplPath string) (string, error) {
 	return retVal, nil
 }
 
+// gpsPos returns the current GPS Position
+func (m *Manager) gpsPos() (gpsd.Position, error) {
```

Comment:
Since the error returned from this function is never used for anything but to check for success, I think we should replace this with a "ok" boolean. Then you could omit the check for "noPos" as well 🙂 

All these error messages could be useful for debugging, so I suggest logging them instead using our debug logger [debug.Printf](https://pkg.go.dev/github.com/la5nta/pat/internal/debug#Printf).

```
func (m *Manager) gpsPos() (pos gpsd.Position, ok bool) {

}
```

I'd be happy to do these last minute tweaks after merging if you prefer. Just say the word 🙂

Reply:
The code currently propagates the return from NextPosTimeout() so this will become quite a bit uglier as would the NoPos checking since gpsFmt is called with a Position.

I've added a couple of debug Printf of the gpsPos error.

Feel free to make the tweaks you suggest if you disagree.

Reply:
Sure, I see what you mean 🙂 

As a general rule of thumb I think it's better to *either* log or return an error, not both. It helps with readability and makes the responsibilities clearer IMO.

I think we should merge this now, and I'll see what can be done with regards to this afterwards.

Thank you 😄